service-registry/src/main/java/org/apache/servicecomb/serviceregistry/consumer/MicroserviceVersions.java

```diff
@@ -193,20 +197,26 @@ private void setInstances(List<MicroserviceInstance> pulledInstances, String rev
 
   private List<MicroserviceInstance> mergeInstances(List<MicroserviceInstance> pulledInstances,
       List<MicroserviceInstance> inUseInstances) {
-    List<MicroserviceInstance> upInstances = pulledInstances
-        .stream()
-        .collect(Collectors.toList());
+    List<MicroserviceInstance> upInstances = new ArrayList<>(pulledInstances);
```

Comment:
Can you describe why we need to add this feature? I think this logic is quite a bit confusing. 1. when service center is down, we can't get any instances; 2. service center status has no relationship with microservice instance status. Now no instances will beremoved when no instances queried from service center.

Reply:
when service center status changes from down to up. It may take some time for provider to registe itself in the service center. so before provider registes itself, the instances get from service center is wrong.